scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/TaskDBUtils.java

```diff
@@ -142,22 +145,24 @@ public Integer executeWork(Session session) {
     }
 
     @NotNull
-    private static Query getQuery(Session session, DBTaskDataParameters params, Set<TaskStatus> taskStatuses, boolean hasUser, boolean hasTag, boolean hasDateFrom, boolean hasDateTo, String queryPrefix) {
+    private static Query getQuery(Session session, DBTaskDataParameters params, Set<TaskStatus> taskStatuses,
+                                  boolean hasUser, boolean hasTag, boolean hasDateFrom, boolean hasDateTo,
+                                  SortSpecifierContainer sortParams, String queryPrefix) {
         StringBuilder queryString = new StringBuilder(queryPrefix);
-        queryString.append(getQueryFilteringExpression(hasUser, hasTag, hasDateFrom, hasDateTo));
-
+        queryString.append(getQueryFilteringExpression(hasUser, hasTag, hasDateFrom, hasDateTo, sortParams));
         Query query = session.createQuery(queryString.toString());
 
-        setQueryParameters(taskStatuses, hasUser, hasTag, hasDateFrom, hasDateTo, query, params);
+        setQueryParameters(taskStatuses, hasUser, hasTag, hasDateFrom, hasDateTo,
+                query, params);
 
         return query;
     }
 
-    private static StringBuilder getQueryFilteringExpression(boolean hasUser, boolean hasTag, boolean hasDateFrom, boolean hasDateTo) {
+    private static StringBuilder getQueryFilteringExpression(boolean hasUser, boolean hasTag,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "getQueryFilteringExpression" is 14 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)